File: app/src/main/java/me/rei_m/hyakuninisshu/domain/karuta/model/Exam.java
package me.rei_m.hyakuninisshu.domain.karuta.model;

import android.support.annotation.NonNull;

import java.util.Date;
import java.util.List;

import me.rei_m.hyakuninisshu.domain.AbstractEntity;

public class Exam extends AbstractEntity<Exam, ExamIdentifier> {

    public Exam(@NonNull ExamIdentifier identifier,
                @NonNull Date tookExamDate,
                int totalQuizCount,
                @NonNull List<KarutaIdentifier> wrongKarutaIdList) {
        super(identifier);
        this.tookExamDate = tookExamDate;
        this.totalQuizCount = totalQuizCount;
        this.wrongKarutaIdList = wrongKarutaIdList;
    }

    public final Date tookExamDate;

    public final int totalQuizCount;

    public final List<KarutaIdentifier> wrongKarutaIdList;
}

File: domain/src/main/java/me/rei_m/hyakuninisshu/domain/question/model/Exam.kt
/*
 * Copyright (c) 2025. Rei Matsushita
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is
 * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
 * the License for the specific language governing permissions and limitations under the License.
 */

package me.rei_m.hyakuninisshu.domain.question.model

import me.rei_m.hyakuninisshu.domain.AbstractEntity
import java.util.Date

/**
 * 百人一首の力試し.
 *
 * @param id ID
 * @param tookDate 受験日
 * @param result 力試し結果
 */
class Exam(
    id: ExamId,
    val tookDate: Date,
    val result: ExamResult,
) : AbstractEntity<ExamId>(id) {
    override fun toString() = "Exam(id=$identifier, tookDate=$tookDate, result=$result)"
}
